src/lib/tecs/TECS.hpp

```diff
@@ -49,351 +49,616 @@
+	void set_altitude_error_time_constant(float time_const) { _control_param.altitude_error_gain = 1.0f / math::max(time_const, 0.1f);; };
 
-	float STE_rate() { return _SPE_rate + _SKE_rate; }
+	void set_equivalent_airspeed_max(float airspeed) { _equivalent_airspeed_max = airspeed; }
+	void set_equivalent_airspeed_min(float airspeed) { _equivalent_airspeed_min = airspeed; }
+	void set_equivalent_airspeed_trim(float airspeed) { _equivalent_airspeed_trim = airspeed; }
 
-	float STE_rate_setpoint() { return _STE_rate_setpoint; }
+	void set_pitch_damping(float damping) { _control_param.pitch_damping_gain = damping; }
+	void set_vertical_accel_limit(float limit) { _reference_param.vert_accel_limit = limit; _control_param.vert_accel_limit = limit; };
 
-	float SEB() { return _SPE_estimate * _SPE_weighting - _SKE_estimate * _SKE_weighting; }
+	void set_speed_weight(float weight) { _control_param.pitch_speed_weight = weight; };
+	void set_airspeed_error_time_constant(float time_const) { _control_param.airspeed_error_gain = 1.0f / math::max(time_const, 0.1f); };
 
-	float SEB_setpoint() { return _SPE_setpoint * _SPE_weighting - _SKE_setpoint * _SKE_weighting; }
+	void set_throttle_damp(float throttle_damp) { _control_param.throttle_damping_gain = throttle_damp; };
+	void set_throttle_slewrate(float slewrate) { _control_param.throttle_slewrate = slewrate; };
 
-	float SEB_rate() { return _SPE_rate * _SPE_weighting - _SKE_rate * _SKE_weighting; }
+	void set_roll_throttle_compensation(float compensation) { _control_param.load_factor_correction = compensation; };
+	void set_load_factor(float load_factor) { _control_param.load_factor = load_factor; };
 
-	float SEB_rate_setpoint() { return _SPE_rate_setpoint * _SPE_weighting - _SKE_rate_setpoint * _SKE_weighting; }
+	void set_ste_rate_time_const(float time_const) { _control_param.ste_rate_time_const = time_const; };
 
+	void set_seb_rate_ff_gain(float ff_gain) { _control_param.seb_rate_ff = ff_gain; };
 
 	/**
 	 * Handle the altitude reset
 	 *
 	 * If the estimation system resets the height in one discrete step this
 	 * will gracefully even out the reset over time.
 	 */
-	void handle_alt_step(float delta_alt, float altitude)
+	void handle_alt_step(float altitude, float altitude_rate)
 	{
-		_hgt_setpoint += delta_alt;
+		TECSReferenceModel::AltitudeReferenceState init_state{ .alt = altitude,
+				.alt_rate = altitude_rate};
 
-		// reset height states
-		_vert_pos_state = altitude;
-		_vert_vel_state = 0.0f;
+		// reset altitude reference model.
+		_reference_model.initialize(init_state);
 	}
 
+	float get_pitch_setpoint() {return _control.getPitchSetpoint();}
+	float get_throttle_setpoint() {return _control.getThrottleSetpoint();}
+
+	// // TECS status
+	uint64_t timestamp() { return _update_timestamp; }
+	ECL_TECS_MODE tecs_mode() { return _tecs_mode; }
+
+	static constexpr float DT_DEFAULT = 0.02f;
```

Comment:
I don't think DT_DEFAULT is used anymore, is it?

Reply:
No youre right, is a remnant i had to use, when i compared it still to the old implementation.